lib/streams.js

```diff
@@ -54,3 +54,77 @@ class WriteStream extends Writable {
 }
 
 exports.WriteStream = WriteStream
+
+class ByteStream extends Readable {
+  constructor (core, opts = {}) {
+    super()
+
+    this.core = core
+    this.byteOffset = opts.byteOffset || 0
+    this.byteLength = typeof opts.byteLength === 'number' ? opts.byteLength : -1
+    this.prefetch = typeof opts.prefetch === 'number' ? opts.prefetch : 32
+
+    this.index = 0
+    this.range = null
+  }
+
+  _open (cb) {
+    this._openp().then(cb, cb)
+  }
+
+  async _openp () {
+    if (this.byteLength === -1) {
+      await this.core.update()
+      this.byteLength = this.core.byteLength
```

Comment:
- byteOffset no? (max'ed with 0)

Reply:
i'd just call this open like below and do the same tricks

Reply:
`-1` means max byte length, so unsure if I should apply the minus offset

Reply:
Will add tests, that could show any obvious bug

Reply:
You're right, I thought it was like `byteEnd`